adept/mlflow_logging.py

```diff
@@ -32,12 +42,31 @@ class MlflowLoggingModule(eqx.Module):
     with_mlflow: bool
 
     @staticmethod
-    def create_mlflow_run_in_callback(dummy_arg):
-        def create_mlflow_run(dummy_arg):
-            with mlflow.start_run() as run:
+    def create_mlflow_run_in_callback(dummy_arg, mlflow_kwargs):
+        if mlflow_kwargs is None:
+            mlflow_kwargs = {}
+
+        if "parent_run_id" in mlflow_kwargs:
+            assert isinstance(mlflow_kwargs["parent_run_id"], MLRunId)
+
+        kwarg_keys = list(mlflow_kwargs.keys())
+        for k in kwarg_keys:
+            assert k in MLFLOW_KWARG_NAMES
+
+        def create_mlflow_run(dummy_arg, maybe_parent_run_id=None):
+            nested = maybe_parent_run_id is not None
+            parent_run_id = str(maybe_parent_run_id) if nested else None
```

Comment:
Why the maybe?

What is wrong with

```suggestion
        def create_mlflow_run(dummy_arg, parent_run_id=None):
            parent_run_id = str(parent_run_id) if parent_run_id else None
```